runtime/onert/core/include/ir/Index.h

```diff
@@ -29,6 +29,8 @@ using OperationIndex = ::onert::util::Index<uint32_t, OperationIndexTag>;
 
 struct OperandIndexTag;
 using OperandIndex = ::onert::util::Index<uint32_t, OperandIndexTag>;
+static int const OPTIONAL_OPERAND = 0xFFFFFFFE;
```

Comment:
I thought "optional operand" was to refer an operand that can be specified or not.
If I got this correct, `OPTIONAL_OPERAND` sounds a little awkward to me. As far as I understood `0xFFFFFFFE` is to express an operand whose value is not given for an optional operand. So I think this should be `UNSPECIFIED_OPTIONAL_OPERAND`, `EMPTY_OPERAND` or so. Or just reuse `UNDEFINED`(I still think `UNDEFINED` would be enough, but I am also good with distinguishing those two).

Reply:
@wateret tensorflow lite uses `-1` to represent when an input is not given. In addition, tensorflow lite uses term `optional` in several location to represent tensor index is given as `-1`

```c#
table Operator {
  ...
  // Optional input are indicated by -1.
  inputs:[int];
```

```c
#define kTfLiteOptionalTensor (-1)
```

```c
if (tensor_index != kTfLiteOptionalTensor) {
```

To be perfectly honest, I had no trouble to understand and use the same name. However, if you think it is not clear, I may change the name.

- `UNSPECIFIED_OPTIONAL_OPERAND` is too looong.

- `EMPTY_OPERAND` is ambiguous like nullptr to string vs. empty string ("").

- `UNDEFINED`, I don't like two reason:

1. `UNDEFINED` is usually used when it is never initialized (e.g. ECMAScript, ...)
I don't think optional is this case.

2. base class Index says it is invalid if it is `UNDEFINED`. However, I think optional input tensor (= `-1`) is valid.
```
bool valid() const { return _index != UNDEFINED; }
```

If you'd like me to choose one of them, I would choose `UNDEFINED` because it is short, and requires little change.

Reply:
> UNDEFINED is usually used when it is never initialized (e.g. ECMAScript, ...)
> I don't think optional is this case.

In ECMAScript, for example,

```js
function foo(a, b)
{
  console.log(a);
  console.log(b);
}

foo(1);
```

And the result is:

```
1
undefined
```

So I think it is not that unnatural.

Reply:
@glistening Talked in offline, I understood your concern, and we may revise like your previous version when necessary :)